tests/sentry/search/events/builder/test_metrics.py

```diff
@@ -2035,6 +2031,34 @@ def test_run_query_with_on_demand_count(self):
             ],
         )
 
+    # Once we delete the current spec version this test will fail and we can delete it
+    def test_on_demand_builder_with_new_spec(self):
+        field = "count()"
+        query = "transaction.duration:>0"
+        spec = OnDemandMetricSpec(field=field, query=query, spec_type=MetricSpecType.DYNAMIC_QUERY)
```

Comment:
Even though, we do not pass a spec version with the latest spec version, when we use query builder with the feature flag it will pick up the latest spec version.